common/src/test/java/io/netty/util/concurrent/DefaultThreadFactoryTest.java

```diff
@@ -0,0 +1,228 @@
+        assertThat(capturedUh.get(), equalTo((Thread.UncaughtExceptionHandler) sister));
+    }
+
+    @Test
+    public void testDescendantThreadGroups() throws InterruptedException {
+        final SecurityManager current = System.getSecurityManager();
+
+        // install security manager that only allows parent thread groups to mess with descendant thread groups
+        System.setSecurityManager(new SecurityManager() {
+            @Override
+            public void checkAccess(ThreadGroup g) {
+                final ThreadGroup source = Thread.currentThread().getThreadGroup();
+
+                if (source != null) {
+                    if (!source.parentOf(g)) {
+                        throw new SecurityException("source group is not an ancestor of the target group");
+                    }
+                    super.checkAccess(g);
+                }
+            }
+
+            // so we can restore the security manager at the end of the test
+            @Override
+            public void checkPermission(Permission perm) {
+            }
+        });
+
+        // holder for the thread factory, plays the role of a global singleton
+        final AtomicReference<DefaultThreadFactory> factory = new AtomicReference<DefaultThreadFactory>();
+        final AtomicInteger counter = new AtomicInteger();
+        final Runnable task = new Runnable() {
+            @Override
+            public void run() {
+                counter.incrementAndGet();
+            }
+        };
+
+        // create the thread factory, since we are running the thread group brother, the thread
+        // factory will now forever be tied to that group
+        // we then create a thread from the factory to run a "task" for us
+        final Thread first = new Thread(new ThreadGroup("brother"), new Runnable() {
+            @Override
+            public void run() {
+                factory.set(new DefaultThreadFactory(this.getClass()));
+                final Thread t = factory.get().newThread(task);
+                t.start();
+                try {
+                    t.join();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
+                }
+            }
+        });
+        first.start();
+        first.join();
+
+        // now we will use factory again, this time from a sibling thread group sister
+        // if DefaultThreadFactory is "sticky" about thread groups, a security manager
+        // that forbids sibling thread groups from messing with each other will strike this down
+        final Thread second = new Thread(new ThreadGroup("sister"), new Runnable() {
+            @Override
+            public void run() {
+                final Thread t = factory.get().newThread(task);
+                t.start();
+                try {
+                    t.join();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2186)